hamilton/execution/graph_functions.py

```diff
@@ -214,57 +215,102 @@ def dfs_traverse(
+                        result = adapter.call_lifecycle_method_sync(
+                            "do_node_execute",
                             run_id=run_id,
                             node_=node_,
                             kwargs=kwargs,
-                            success=success,
-                            error=error,
-                            result=result,
                             task_id=task_id,
                         )
-                    except Exception:
-                        message = create_error_message(kwargs, node_, "[post-node-execute]")
-                        logger.exception(message)
-                        raise
-            ##### THERE #####
-
+                    else:
+                        result = node_(**kwargs)
+                    
+                    return error, result, success, pre_node_execute_errored
+                
+                except Exception as e:
+                    success = False
+                    error = e
+                    step = "[pre-node-execute]" if pre_node_execute_errored else ""
+                    message = create_error_message(kwargs, node_, step)
+                    logger.exception(message)
+                    raise
+                finally:
+                    if not pre_node_execute_errored and adapter.does_hook(
+                        "post_node_execute", is_async=False
+                    ):
+                        try:
+                            adapter.call_all_lifecycle_hooks_sync(
+                                "post_node_execute",
+                                run_id=run_id,
+                                node_=node_,
+                                kwargs=kwargs,
+                                success=success,
+                                error=error,
+                                result=result,
+                                task_id=task_id,
+                            )
+                        except Exception:
+                            message = create_error_message(kwargs, node_, "[post-node-execute]")
+                            logger.exception(message)
+                            raise
+                    return error, result, success, pre_node_execute_errored
+
+                ##### THERE #####
+            if adapter.does_method("do_remote_execute", is_async=False):
+                 error, result, success, pre_node_execute_errored = adapter.call_lifecycle_method_sync(
+                    "do_remote_execute",
+                    execute_lifecycle_for_node = execute_lifecycle_for_node(
+                        node_ = node_,
+                        adapter = adapter,
+                        run_id = run_id,
+                        task_id = task_id,
+                        kwargs = kwargs,
+                        success=success,
+                        pre_node_execute_errored=pre_node_execute_errored,
+                        error=error,
+                        result=result),
```

Comment:
@jernejfrank I think you're actually calling the function here -- what I think we want is to pass a pointer to the function, along with the arguments for it.

Reply:
Some of these could be bound via functools.partial

Reply:
> @jernejfrank I think you're actually calling the function here -- what I think we want is to pass a pointer to the function, along with the arguments for it.

@skrawcz thanks a lot for pointing that out, I wasn't exactly sure where the arguments need to be passed into the Callable.



> Some of these could be bound via functools.partial

We discussed that option with @elijahbenizzy , but agreed that we both find inline functions clearer to read vs partial. Having said that, I am happy to change that since I have never done partials and would learn something new!

Reply:
yep so you can either:

1. Bind some of the arguments here -- i.e. the ones that don't depend on the inputs to the node function callable.
2. Or pass in a dictionary of those arguments for the function.

In either case, you need to construct the right kwargs for the `.remote(...)` call.